opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -18,7 +18,17 @@
 /**
  * Tracer is a simple, thin interface for Span creation and propagation across arbitrary transports.
  */
-public interface Tracer extends ActiveSpanSource {
+public interface Tracer {
+
+    /**
+     * @return the current {@link ScopeManager}, which may be a noop but may not be null.
+     */
+    ScopeManager scopeManager();
+
+    /**
+     * Set a new {@link ScopeManager}, presumably as part of initialization.
+     */
+    void setScopeManager(ScopeManager scopeManager);
```

Comment:
I still think `setScopeManager` is dangerous, and we should leave it out for now. This is init state, not runtime state, and we should add a `TracerBuilder` concept to assist with the various init states that are showing up (the `ServiceName` attribute, for example).

Reply:
(I think we agree about this)